thunder/core/symbol.py

```diff
@@ -650,35 +652,18 @@ def has_tags(bsym: BoundSymbol, tags: set[OpTags]) -> bool:
 
 # Wrapper class that hashes and equates only the right hand side of a BoundSymbol for CSE.
 # That is to say, its symbol, args, and kwargs, but not its output.
-@dataclass
-class BoundSymbolRHS:
```

Comment:
Frozen dataclass and namedtuples are very similar. Dataclasses are used in other parts of the project and I don't think there's a need to change it here. Can you please explain your motivation and what limitations in dataclasses do you see?

Reply:
True! My main idea behind using a tuple based data structure is its immutable property. `NamedTuple` is very similar to dataclass as noted in the original [PEP](https://peps.python.org/pep-0557):
>  Data Classes can be thought of as “mutable namedtuples with defaults”. 

In our case we use the dataclass with the `frozen` tag set to true which make them immutable and even closer to `NamedTuple`. However, tuples offer the advantage of having a small memory footprint. The only practical limitation I see in using a frozen dataclass is the hack required to change a value whereas namedtuple has a convenient method for it. In any case I recognize that this is a weak point as if we want a mutable dataclass then we shouldn't declare it as frozen and hack around.

With that said, I recognize that the difference in our use case is minimal, and if it is better for readability to use a frozen dataclass instead, it's ok for me